python/ray/serve/_private/utils.py

```diff
@@ -535,3 +545,27 @@ def record_serve_tag(key: str, value: str):
         )
 
     record_extra_usage_tag(serve_telemetry_tag_map[key], value)
+
+
+def _extract_self_if_method_call(args: List[Any], func: Callable) -> Optional[object]:
+    """Check if this is a method rather than a function.
+
+    Does this by checking to see if `func` is the attribute of the first
+    (`self`) argument under `func.__name__`. Unfortunately, this is the most
+    robust solution to this I was able to find. It would also be preferable
+    to do this check when the decorator runs, rather than when the method is.
+
+    Returns the `self` object if it's a method call, else None.
+
+    Arguments:
+        args (List[Any]): arguments to the function/method call.
+        func: the unbound function that was called.
```

Comment:
```suggestion
        args: arguments to the function/method call.
        func: the unbound function that was called.
```
nit (on myself :)) -- redundant with type annotations